Tests/SwiftLexicalLookupTest/NameLookupTests.swift

```diff
@@ -545,17 +550,35 @@ final class testNameLookup: XCTestCase {
 
         9️⃣class d {}
 
-        let x = 0️⃣d
+        let 🔟a = 0️⃣d
         """,
       references: [
         "3️⃣": [.fromFileScope(expectedNames: ["1️⃣"])],
-        "4️⃣": [],
-        "5️⃣": [],
-        "6️⃣": [],
+        "4️⃣": [.fromFileScope(expectedNames: ["2️⃣"])],
+        "5️⃣": [.fromFileScope(expectedNames: ["7️⃣"])],
+        "6️⃣": [.fromFileScope(expectedNames: ["9️⃣"])],
         "0️⃣": [.fromFileScope(expectedNames: ["9️⃣"])],
       ],
       expectedResultTypes: .all(ClassDeclSyntax.self, except: ["8️⃣": IdentifierPatternSyntax.self]),
       config: LookupConfig(fileScopeHandling: .codeBlock)
     )
   }
+  
+  func testDeclarationAvailabilityInCodeBlock() {
+    assertLexicalNameLookup(
+      source: """
+        func x {
+          1️⃣class A {}
+        
+          let a = 2️⃣A()
+        
+          3️⃣class A {}
+        }
+        """,
+      references: [
+        "2️⃣": [.fromScope(CodeBlockSyntax.self, expectedNames: ["1️⃣", "3️⃣"])]
+      ],
+      expectedResultTypes: .all(ClassDeclSyntax.self)
+    )
+  }
```

Comment:
I simply [removed `accessibleAfter` from `LookupName.declaration`](https://github.com/swiftlang/swift-syntax/blob/b85751b496b82a41953343b3183595255453ec87/Sources/SwiftLexicalLookup/LookupName.swift#L21). I don't think it should've been there in the first place, so now it should work. It will require a bit more work with `guard` handling in #2748, but I'll fix it there once we merge this PR.

Reply:
Oh, I missed that. Thanks for the explanation.